src/code/z_actor.c

```diff
@@ -1781,56 +1798,81 @@ PosRot* Actor_GetWorldPosShapeRot(PosRot* dest, Actor* actor) {
     return dest;
 }
 
-f32 func_800B82EC(Actor* actor, Player* player, s16 angle) {
-    f32 temp_f12;
-    s16 temp_v0 = BINANG_SUB(BINANG_SUB(actor->yawTowardsPlayer, 0x8000), angle);
-    s16 yaw = ABS_ALT(temp_v0);
+/**
+ * Returns the squared xyz distance from the actor to Player.
+ *
+ * This distance will be adjusted smaller if Player is already targeting an actor. The more Player is facing the actor,
+ * the smaller the distance is adjusted.
+ */
+f32 Target_GetAdjustedDistSq(Actor* actor, Player* player, s16 playerShapeYaw) {
+    f32 adjDistSq;
+    s16 yawDiff;
+
+    // The yaw, with player as the origin, from where player is facing to where the actor is positioned
+    yawDiff = ABS_ALT(BINANG_SUB(BINANG_SUB(actor->yawTowardsPlayer, 0x8000), playerShapeYaw));
 
     if (player->targetedActor != NULL) {
-        if ((yaw > 0x4000) || ((actor->flags & ACTOR_FLAG_CANT_LOCK_ON))) {
+        if ((yawDiff > (0x10000 / 4)) || (actor->flags & ACTOR_FLAG_CANT_LOCK_ON)) {
```

Comment:
I don't really see the advantage of breaking it up into quarters. I don't think the 4 here has any significant meaning?  If that number were to change, then the `0x4000` and `0x8000` in the `adjDistSq` calc would need to be adjusted too.